attest/certification_test.go

```diff
@@ -274,3 +276,72 @@ func testKeyCertification(t *testing.T, tpm *TPM) {
 		})
 	}
 }
+
+func TestKeyActivationTPM20(t *testing.T) {
+	sim, tpm := setupSimulatedTPM(t)
+	defer sim.Close()
+
+	ak, err := tpm.NewAK(nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	akAttestParams := ak.AttestationParameters()
+	pub, err := tpm2.DecodePublic(akAttestParams.Public)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if pub.Type != tpm2.AlgRSA {
+		t.Fatal("non-RSA verifying key")
+	}
+
+	eks, err := tpm.EKs()
+	if err != nil {
+		t.Fatalf("unexpected error retrieving EK from tpm: %v", err)
+	}
+
+	if len(eks) == 0 {
+		t.Fatal("expected at least one EK")
+	}
+
+	pk := &rsa.PublicKey{E: int(pub.RSAParameters.Exponent()), N: pub.RSAParameters.Modulus()}
+	hash, err := pub.RSAParameters.Sign.Hash.Hash()
+	if err != nil {
+		t.Fatal(err)
+	}
+	verifyOpts := VerifyOpts{
+		Public: pk,
+		Hash:   hash,
+	}
+
+	pubName, err := pub.Name()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	activateOpts := ActivateOpts{
+		EK:   eks[0].Public,
+		Hash: pubName.Digest,
+	}
+
+	sk, err := tpm.NewKey(ak, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rand := mathrand.New(mathrand.NewSource(123456))
```

Comment:
why does this need to be a consistent value and not use a regular rand.Reader? Can you add a comment at the very least?

Reply:
Will leave as a regular rand.Reader as it does not have to be a fixed value